script.js

```diff
@@ -0,0 +1,24 @@
+const carousel = document.querySelector('.carousel');
+
+function changeSlide(param) {
+    const currentSlideIndex = Number(getComputedStyle(carousel).getPropertyValue('--slide-index'));
```

Comment:
I'd say the `getComputedStyle` logic here is too much -- I'd recommend simplifying a bit:

```
const ui = {
  carousel: document.querySelector('.carousel'),
  slides: document.querySelectorAll('.slide'),
  slidesCounter: document.querySelector('.slides-counter'),
  slideIndex: 0
}
```

And then just:

```
function nextSlide() {
  if (ui.slideIndex === ui.slides.length - 1) return;

  // rest of the logic
}

function prevSlide() {
  if (ui.slideIndex === 0) return;

  // rest of the logic
}

function updateCount() {
  ui.slidesCounter.textContent = `${ui.slideIndex}/${ui.slides.length}`
}
```